File: 백준/Silver/1652. 누울 자리를 찾아라/누울 자리를 찾아라.py
N = int(input())
lst = [input() for i in range(N)]

row = 0
column = 0

for i in range(N):
    count = 0 
    for j in range(N):
        if lst[i][j] == '.':
            count+=1
        else:
            if count >=2:
                row+=1
            count = 0
    if count >=2:
        row+=1
        
for j in range(N):
    count = 0 
    for i in range(N):
        if lst[i][j] == '.':
            count+=1
        else:
            if count >=2:
                column+=1
            count = 0        
    if count >=2:
        column+=1
print(row,column)  

File: 백준/Silver/1652. 누울 자리를 찾아라/누울 자리를 찾아라.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(br.readLine());

        char [][] lst = new char [N][N];

        for(int i = 0; i < N; i++){
            String str = br.readLine();
            for(int j = 0; j < N; j++){
                lst[i][j] = str.charAt(j);
            }
        }

        int row = 0;

        for(int i = 0; i < N; i++){
            Stack<Character> st = new Stack<>();
            boolean flag = true;

            for(int j = 0; j < N; j++){
                if(lst[i][j] == '.'){
                    st.push(lst[i][j]);
                    if(st.size() == 2 && flag){
                        row ++;
                        flag = false;
                    }
                }else{
                    st.clear();
                    flag = true;
                }
            }
        }

        int col = 0;

        for(int i = 0; i < N; i++){
            Stack<Character> st = new Stack<>();
            boolean flag = true;

            for(int j = 0; j < N; j++){
                if(lst[j][i] == '.'){
                    st.push(lst[j][i]);
                    if(st.size() >= 2 && flag){
                        col ++;
                        flag = false;
                    }
                }else{
                    st.clear();
                    flag = true;
                }
            }
        }

        System.out.println(row + " " + col);
  
    }
}
